merlin/spec/specification.py

```diff
@@ -117,6 +117,7 @@ def process_spec_defaults(self):
         MerlinSpec.fill_missing_defaults(self.batch, defaults.BATCH["batch"])
 
         # fill in missing step section defaults within 'run'
+        defaults.STUDY_STEP_RUN["shell"] = self.batch["shell"]
```

Comment:
Wondering how we had things working before without this...

In any case it looks like it'll never cause a KeyError. Looks good to me

Reply:
The shell wasn't set in the step here before.